Sources/SwiftParser/ParseSourceFile.swift

```diff
@@ -62,12 +70,42 @@ extension Parser {
   ///            how far the parser looked ahead while parsing a node, which is
   ///            necessary to construct an ``IncrementalParseTransition`` for a
   ///            subsequent incremental parse
+  @available(*, deprecated, message: "Use the new version that returns a struct instead.")
   public static func parseIncrementally(
     source: String,
     parseTransition: IncrementalParseTransition?
   ) -> (tree: SourceFileSyntax, lookaheadRanges: LookaheadRanges) {
+    let result: IncrementalParseResult = parseIncrementally(source: source, parseTransition: parseTransition)
+    return (result.tree, result.lookaheadRanges)
+  }
+
+  /// Parse the source code in the given string as Swift source file with support
+  /// for incremental parsing.
+  ///
+  /// When parsing a source file for the first time, invoke `parseIncrementally`
+  /// with `parseTransition: nil`. This returns the initial tree as well as
+  /// ``LookaheadRanges``. If an edit is made to the source file, an
+  /// ``IncrementalParseTransition`` can be constructed from the initial tree
+  /// and its ``LookaheadRanges``. When invoking `parseIncrementally` again with
+  /// the post-edit source and that parse transition, the parser will re-use
+  /// nodes that haven’t changed.
+  ///
+  /// - Parameters:
+  ///   - source: The source code to parse
+  ///   - parseTransition: If a similar source file has already been parsed, the
+  ///     ``IncrementalParseTransition`` that contains the previous tree as well
+  ///     as the edits that were performed to it.
+  /// - Returns: The ``IncrementalParseResult``.
+  public static func parseIncrementally(
```

Comment:
okay this will break as is, as the compiler cannot figure out what function to use. 

We could, for now, say that it should use `@_disfavoredOverload` on the new one if we want.

Then people can transition